### DISCUSSION TITLE
Uncaught TypeError: el.matches is not a function

### DISCUSSION THREAD

**Discussion by ghost** [alpinejs/alpine#1844] [1. Help]
I made a few HTML pages with Tailwind and Alpine, works fine. Now trying to integrate with Laravel and Laravel Mix and everything breaks.

When loading the first error which pops up is:

`Uncaught TypeError: el.matches is not a function` coming from the generated JS bundle in the second line of this function (not my code):

```js
function closestRoot(el) {
  if (rootSelectors().some((selector) => el.matches(selector)))
    return el;
  if (!el.parentElement)
    return;
  return closestRoot(el.parentElement);
}
```

In my app.js I have the following:

```js
import Alpine from 'alpinejs'
window.Alpine = Alpine
Alpine.start()
```

Alpine.start() seems to cause the error. and all my subsequent JS written below the import Alpine.start() line breaks too.

**Comment by SimoTod** [comment]
Do you have an x-data attribute on your html tag by any chance? If so, there's a bug causing alpine to call el.matches on document (which is a node and not an element) and triggering that error.
There's already a fix in master but it hasn't been released yet.

**Reply by ghost** [reply]
Fuck, yes I have. Good catch. Will have a look.

Also I was using Alpine 2.x I just realized, trying to make everything work on 3.x now.

**Comment by ghost** [comment]
Is indeed fixed with self-building the file. Any idea when a new release will be available so I can `npm update`?

**Reply by SimoTod** [reply]
No idea. Caleb is the maintainer of the project, I just send the odd PR when I can see something I can fix.

**Comment by ghost** [comment]
@calebporzio Any ETA on a new tagged version?
